packages/material-ui/src/styles/experimentalStyled.d.ts

```diff
@@ -216,7 +216,7 @@ export interface CreateMUIStyled<Theme extends object = DefaultTheme> {
       theme?: Theme;
       as?: React.ElementType;
       sx?: SxProps<Theme>;
-      styleProps?: Record<string, unknown>;
+      styleProps?: Record<string, any>;
```

Comment:
Why do we need the change here? This PR is only touching a single component so let's not change something that's affecting hundreds of components. If we need to cast something to `any` then let's make it explicit.

Reply:
fixed